src/util_test.cc

```diff
@@ -428,3 +428,13 @@ TEST(ElideMiddle, ElideInTheMiddle) {
   EXPECT_EQ("012...789", elided);
   EXPECT_EQ("01234567...23456789", ElideMiddle(input, 19));
 }
+
+TEST(ElideMiddle, NothingToElideMiddleWithEscape) {
+  string input = "Nothing \33[m string.";
+  EXPECT_EQ("Nothing \33[m string.", ElideMiddle(input, 80));
+}
+
+TEST(ElideMiddle, ElideMiddleWithEscape) {
+  string input = "0\33[m1234567890123456789";
+  EXPECT_EQ("0\33[m123...\33[m789", ElideMiddle(input, 10));
```

Comment:
What happens when the ANSI escape is inside the elided part?

Reply:
All ANSI escapes are always output.

Reply:
Hence the duplicate `\33[m` part on the second half.

Reply:
Hm ... why duplicate it?

Reply:
It grabs the escape sequences from the elided part. I haven't dug into the issue I'm seeing here yet, but fixing it will probably make doing it better easier.